dms-persistence/src/main/kotlin/team/aliens/dms/persistence/vote/mapper/VotingTopicMapper.kt

```diff
@@ -0,0 +1,46 @@
+package team.aliens.dms.persistence.vote.mapper
+
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Component
+import team.aliens.dms.domain.vote.model.VotingTopic
+import team.aliens.dms.persistence.GenericMapper
+import team.aliens.dms.persistence.manager.repository.ManagerJpaRepository
+import team.aliens.dms.persistence.vote.entity.VotingTopicJpaEntity
+
+@Component
+class VotingTopicMapper(
+    private val managerJpaRepository: ManagerJpaRepository
+) : GenericMapper<VotingTopic, VotingTopicJpaEntity> {
+
+    override fun toDomain(entity: VotingTopicJpaEntity?): VotingTopic? {
+        return entity?.let {
+            VotingTopic(
+                id = it.id!!,
+                topicName = it.topicName,
+                description = it.description,
+                startTime = it.startTime,
+                endTime = it.endTime,
+                voteType = it.voteType,
+                managerId = it.manager?.id
+            )
+        }
+    }
+
+    override fun toEntity(domain: VotingTopic): VotingTopicJpaEntity {
+        val manager = if (domain.managerId != null) {
+            managerJpaRepository.findByIdOrNull(domain.managerId!!)
+        } else {
+            null
+        }
+
+        return VotingTopicJpaEntity(
+            id = domain.id,
+            topicName = domain.topicName,
+            description = domain.description,
+            startTime = domain.startTime,
+            endTime = domain.endTime,
+            voteType = domain.voteType,
+            manager = manager
+        )
+    }
```

Comment:
```suggestion
    override fun toDomain(entity: VotingTopicJpaEntity?): VotingTopic? {
        return entity?.let {
            VotingTopic(
                id = it.id!!,
                topicName = it.topicName,
                description = it.description,
                startTime = it.startTime,
                endTime = it.endTime,
                voteType = it.voteType,
                managerId = it.manager!!.id
            )
        }
    }

    override fun toEntity(domain: VotingTopic): VotingTopicJpaEntity {
        val manager =  managerJpaRepository.findByIdOrNull(domain.managerId)
        return VotingTopicJpaEntity(
            id = domain.id,
            topicName = domain.topicName,
            description = domain.description,
            startTime = domain.startTime,
            endTime = domain.endTime,
            voteType = domain.voteType,
            manager = manager
        )
    }
```
비즈니스 로직상 manager 아이디가 null이 될 수 없고 도메인 계층에서는 아이디만 관리되게 설계해야할 것 같아요
VotingTopic 클래스에서도 managerId nullable 선언은 삭제해주세요

Reply:
수정 했습니다